tools/generate-wire.py

```diff
@@ -1077,17 +1089,20 @@ def parse_tlv_file(tlv_field_name):
         continue
 
     if len(parts) == 2:
-        # eg commit_sig,132
-        messages.append(Message(parts[0], Enumtype("WIRE_" + parts[0].upper(), parts[1]), comments))
+        if parts[1] == '$':  # this is a subtype
+            subtypes.append(Subtype(parts[0], comments))
+        else:
```

Comment:
$?  Can we just have a single entry be a subtype?

Note that there's a companion to these changes, which is that the spec itself needs to decide on a format, and lightning-rfc/tools/extract-formats.py needs to be taught to extract it into the form you want here.

Reply:
updated the RFC parser to output desired format here: https://github.com/lightningnetwork/lightning-rfc/pull/597

Reply:
OK, I moved my complaint to that PR :)

Otherwise, excellent.  Be nice to try this out on some internal types, too: autogen FTW.